### PR TITLE
[TH2-1473] Action 'ExecuteJsOnElement' added

### PR DESCRIPTION
**Pull request by danielsamatov**

### CODE REVIEW

File: src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJsOnElement.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/******************************************************************************
```
Comment (Vladimir-Panarin-Exactpro) [inline src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJsOnElement.java]: It is open source project. replace the copyright.

**Review by Vladimir-Panarin-Exactpro — CHANGES_REQUESTED** [review]

**Review by Vladimir-Panarin-Exactpro — CHANGES_REQUESTED** [review]
Copyright in WebJsUtils should also be fixed.

**Review by Vladimir-Panarin-Exactpro — COMMENTED** [review]
@battlemaster Please take a look.

File: src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJS.java [hunk 2]
```diff
@@ -23,66 +30,72 @@
...
-		executeJsCommands(webDriver, commands);
+	protected void executeJsCommands(WebDriver webDriver, List<String> commands, Object... args)
+			throws ScriptExecuteException
+	{
+		WebJsUtils.executeJsCommands(webDriver, commands, args);
+	}
 
+	protected Object[] getJsArguments(WebDriver webDriver, By webLocator, Map<String, String> params)
+	{
 		return null;
```
Comment (std000) [inline src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJS.java]: NPE alert. This value passes into RemoteWebDriver.executeScript, where is not expected null values as varargs. Use new Object[0]

File: src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJsOnElement.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+{
+	private static final Logger logger = LoggerFactory.getLogger(ExecuteJsOnElement.class);
+
+	protected static final String ELEMENT_PARAM = "%Element%";
+	protected static final String ARGUMENT = "arguments[0]";
+
+	@Override
+	protected String getJsScript(Map<String, String> params)
+	{
+		return params.get(JS_COMMANDS).replace(ELEMENT_PARAM, ARGUMENT);
```
Comment (std000) [inline src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJsOnElement.java]: Check params.get(JS_COMMANDS) is empty or null

File: src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJS.java [hunk 4]
```diff
@@ -23,66 +30,72 @@
...
+	protected static final String NOTHING_TO_EXECUTE_MESSAGE = "Nothing to execute";
 	protected static final String DEFAULT_DELIMITER = ";";
 
-
 	@Override
-	public boolean isNeedLocator()
+	public String run(WebDriver webDriver, By webLocator, Map<String, String> params) throws ScriptExecuteException
 	{
-		return false;
+		executeJsCommands(webDriver, splitCommands(getJsScript(params)), getJsArguments(webDriver, webLocator, params));
```
Comment (std000) [inline src/main/java/com/exactprosystems/remotehand/web/actions/ExecuteJS.java:43]: Check that descendants don't return null values, or hadle null correctly

**Review by std000 — COMMENTED** [review]
Also update version in gradle.properties.

**Review by std000 — APPROVED** [review]

**Review by Vladimir-Panarin-Exactpro — APPROVED** [review]

**Pull request merged**